docs/api-reference/expression-estimator.md

```diff
@@ -0,0 +1,181 @@
+﻿### The Expression Language
+
+The language for the expression estimator should be comfortable to a broad range of users. It shares many similarities with some popular languages.
+It is case sensitive, supports multiple types and has a rich set of operators and functions. It is pure functional, in the sense that there are no
+mutable values or mutating operations in the language. It does not have, nor need, any exception mechanism, instead producing NA values when a normal
+value is not appropriate. It is statically typed, but all types are inferred by the compiler.
+
+## Syntax
+
+Syntax for the lambda consists of a parameter list followed by either colon (:) or arrow (=>) followed by an expression.
+The parameter list can be either a single identifier or a comma-separated list of one or more identifiers surrounded by parentheses.
+
+_lambda:_
+
+-    _parameter-list   **:** expression_
+-    _parameter-list   **=>** expression_
+
+_parameter-list:_
+
+- _identifier_
+- **(**  _parameter-names_   **)**
+
+_parameter-names:_
+
+- _identifier_
+- _identifier   **,**  parameter-names_
+
+The expression can use parameters, literals, operators, with-expressions, and functions.
+
+## Literals
+
+- The boolean literals are true and false.
+- Integer literals may be decimal or hexadecimal (e.g., 0x1234ABCD). They can be suffixed with u or U, indicating unsigned,
+as well as l or L, indicating long (Int64). The use of u or U is rare and only affects promotion of certain 32 bit hexadecimal values,
+determining whether the constant is considered a negative Int32 value or a positive Int64 value.
+- Floating point literals use the standard syntax, including exponential notation (123.45e-37). They can be suffixed with
+f or F, indicating single precision, or d or D, indicating double precision. Unlike in C#, the default precision of a
+floating point literal is single precision. To specify double precision, append d or D.
+- Text literals are enclosed in double-quotation marks and support the standard escape mechanisms.
+
+## Operators
+
+The operators of the expression language are listed in the following table, in precendence order. Unless otherwise noted,
```

Comment:
We should mention how the expression handles vector columns. Hopefully we still operate on them :)

Reply:
#Resolved